types/expressions.d.ts

```diff
@@ -0,0 +1,2880 @@
+      $mod: [NumberExpression, NumberExpression];
+    }
+    export interface Multiply {
+      /**
+       * Multiplies numbers to return the product. Accepts any number of argument expressions.
+       *
+       * @see https://docs.mongodb.com/manual/reference/operator/aggregation/multiply/#mongodb-expression-exp.-multiply
+       */
+      $multiply: NumberExpression[];
+    }
+
+    export interface Pow {
+      /**
+       * Raises a number to the specified exponent.
+       *
+       * @see https://docs.mongodb.com/manual/reference/operator/aggregation/pow/#mongodb-expression-exp.-pow
+       */
+      $pow: [NumberExpression, NumberExpression];
+    }
+
+    export interface Round {
+      /**
+       * Rounds a number to to a whole integer or to a specified decimal place.
+       *
+       * @see https://docs.mongodb.com/manual/reference/operator/aggregation/round/#mongodb-expression-exp.-round
+       */
+      $round: [NumberExpression, NumberExpression?];
+    }
+
+    export interface Sqrt {
+      /**
+       * Calculates the square root.
+       *
+       * @see https://docs.mongodb.com/manual/reference/operator/aggregation/sqrt/#mongodb-expression-exp.-sqrt
+       */
+      $sqrt: NumberExpression;
+    }
+
+    export interface Subtract {
+      /**
+       * Returns the result of subtracting the second value from the first. If the two values are numbers, return the difference. If the two values are dates, return the difference in milliseconds. If the two values are a date and a number in milliseconds, return the resulting date. Accepts two argument expressions. If the two values are a date and a number, specify the date argument first as it is not meaningful to subtract a date from a number.
+       *
+       * @see https://docs.mongodb.com/manual/reference/operator/aggregation/subtract/#mongodb-expression-exp.-subtract
+       */
+      $subtract: (NumberExpression | DateExpression)[];
+    }
+
+    export interface Trunc {
+      /**
+       * Truncates a number to a whole integer or to a specified decimal place.
+       *
+       * @see https://docs.mongodb.com/manual/reference/operator/aggregation/trunc/#mongodb-expression-exp.-trunc
+       */
+      $trunc: [NumberExpression, NumberExpression?];
+    }
+
+    export interface Sin {
+      /**
+       * Returns the sine of a value that is measured in radians.
+       *
+       * @see https://docs.mongodb.com/manual/reference/operator/aggregation/sin/#mongodb-expression-exp.-sin
+       */
+      $sin: NumberExpression;
+    }
+
+    export interface Cos {
+      /**
+       * Returns the cosine of a value that is measured in radians.
+       *
+       * @version 4.2
+       * @see https://docs.mongodb.com/manual/reference/operator/aggregation/cos/#mongodb-expression-exp.-cos
+       */
+      $cos: NumberExpression;
+    }
+
+    export interface Tan {
+      /**
+       * Returns the tangent of a value that is measured in radians.
+       *
+       * @version 4.2
```

Comment:
Can we get rid of the `@version` tags? It's a bit confusing because `@version` refers to the MongoDB server rather than the Mongoose version